client/components/timezone-dropdown/index.jsx

```diff
@@ -3,50 +3,100 @@
  * External Dependencies
  */
 import React from 'react';
-import noop from 'lodash/noop';
+import i18n from 'lib/mixins/i18n';
 
 /**
  * Internal dependencies
  */
+import wpcom from 'lib/wp';
 import Dropdown from 'components/select-dropdown';
-import i18n from 'lib/mixins/i18n';
 
-export default React.createClass( {
-	displayName: 'TimezoneDropdown',
+/**
+ * Module variables
+ */
+const { Component, PropTypes } = React;
+const noop = () => {};
+const undocumented = wpcom.undocumented();
+
+class TimezoneDropdown extends Component {
+	constructor() {
+		super();
 
-	propTypes: {
-		selectedZone: React.PropTypes.string,
-		onSelect: React.PropTypes.func
-	},
+		// bound methods
+		this.onSelect = this.onSelect.bind( this );
 
-	getDefaultProps() {
-		return {
-			onSelect: noop
+		this.state = {
+			timezones: [ { label: '', value: '' } ]
 		};
-	},
-
-	getTimezoneNames() {
-		return i18n.moment.tz.names().map( zone => {
-			return ( {
-				label: zone,
-				value: zone
-			} );
+	}
+
+	componentWillMount() {
+		undocumented.timezones( ( err, zones ) => {
+			if ( err ) {
+				return console.log( err );
+			}
+
+			let timezones = [];
+
+			for ( let continent in zones.timezones_by_continent ) {
+				let cities = zones.timezones_by_continent[ continent ];
+				cities = [ {
+					label: continent,
+					value: continent,
+					isLabel: true
+				} ].concat( cities, [ null ] );
+
+				timezones = timezones.concat( cities );
+			}
+
+			timezones = timezones.concat( [
+				{
+					label: 'UTC',
+					value: 'UTC',
+					isLabel: true
+				},
+				{
+					label: 'UTC',
+					value: 'UTC'
+				},
+				null
+			] );
+
+			timezones = timezones.concat( [
+				{
+					label: i18n.translate( 'Manual Offsets' ),
+					value: 'manual-offsets',
+					isLabel: true
+				} ], zones.manual_utc_offsets );
+
+			this.setState( { timezones } );
 		} );
-	},
+	}
 
 	onSelect( zone ) {
-		this.setState( { selectedZone: zone.value } );
-		this.props.onSelect( zone.value );
-	},
+		this.props.onSelect( zone );
+	}
 
 	render() {
 		return (
 			<Dropdown
 				className="timezone-dropdown"
-				options={ this.getTimezoneNames() }
+				valueLink={ this.props.valueLink }
+				options={ this.state.timezones }
 				selectedText={ this.props.selectedZone }
 				onSelect={ this.onSelect }
 			/>
 		);
-	},
-} );
+	}
+};
+
+TimezoneDropdown.defaultProps = {
+	onSelect: noop
+};
+
+TimezoneDropdown.propTypes = {
+	selectedZone: PropTypes.string,
+	onSelect: PropTypes.func
+}
```

Comment:
I've followed this doc:

https://facebook.github.io/react/docs/reusable-components.html#es6-classes

> Another difference is that propTypes and defaultProps are defined as properties on the constructor instead of in the class body.